lib/widget/create_room_page.dart

```diff
@@ -60,11 +61,16 @@ class CreateRoomPage extends StatelessWidget {
 
   Widget _createRoomButton(BuildContext context, Store<GameModel> store) => new RaisedButton(
         child: new Text(AppLocalizations.of(context).createRoom, style: buttonTextStyle),
-        onPressed: () {
-          FormState enterNameState = Keys.createRoomPageNameKey.currentState;
-          if (enterNameState.validate()) {
-            enterNameState.save();
-            store.dispatch(new CreateRoomAction());
+        onPressed: () async {
+          var connectivityResult = await (new Connectivity().checkConnectivity());
```

Comment:
If this check is slow, the user could probably see nothing happen when they tap the button, and try to tap it many times.

Maybe we should do the internet check inside the `CreateRoomAction` (which uses `withRequest()` to display a loading spinner).